packages/types/src/serde.ts

```diff
@@ -92,7 +92,7 @@ export type SdkStream<BaseStream> = BaseStream & SdkStreamMixin;
  * with the SdkStreamMixin helper methods.
  */
 export type WithSdkStreamMixin<T, StreamKey extends keyof T> = {
-  [key in keyof T]: T[key] extends T[StreamKey] ? SdkStream<T[StreamKey]> : T[key];
+  [key in keyof T]: [T[key]] extends [T[StreamKey]] ? SdkStream<T[StreamKey]> : T[key];
```

Comment:
```suggestion
  [key in keyof T]: key extends StreamKey ? SdkStream<T[StreamKey]> : T[key];
```

Reply:
```typescript
type Stream = {} | undefined;

type Data = {
    undef: undefined
    num: number
    str: string
    stream: Stream
}

type Mixin<T> = T & {
    someAddedMethod(): void
}

// for v1/v2, the match is on the type of the key indicated. This is too wide if the Stream type includes undefined.
type ApplyConditionalTypeMappingV1<T, Key extends keyof T> = {
    [key in keyof T]: T[key] extends T[Key] ? Mixin<T[Key]> : T[key];
}

type ApplyConditionalTypeMappingV2<T, Key extends keyof T> = {
    [key in keyof T]: [T[key]] extends [T[Key]] ? Mixin<T[Key]> : T[key];
}

// this applies the type mixin only to the specified key.
type ApplyConditionalTypeMappingV3<T, Key extends keyof T> = {
    [key in keyof T]: key extends Key ? Mixin<T[Key]> : T[key];
}

const data: ApplyConditionalTypeMappingV3<Data, 'stream'> = {
    num: 1,
    str: 'a',
    stream: undefined,
    undef: undefined,
}
```

Reply:
Agreed that this is more robust

Reply:
Updated docs, however this does change the behaviour and I'm not sure if the change from matching multiple keys to only a single key is expected (even though the signature of the mixin definitely indicates it only should match one key).

Reply:
Yes, there will only be one key. I believe this is guaranteed by Smithy.